Source: https://github.com/manugarg/pacparser/issues/194#issuecomment-2262030966

Fix -Wincompatible-pointer-types error with gcc 14.

```
jsapi.c: In function 'JS_ConvertArgumentsVA':
jsapi.c:96:35: error: passing argument 5 of 'TryArgumentFormatter' from
incompatible pointer type [-Wincompatible-pointer-types]
   96 | #define JS_ADDRESSOF_VA_LIST(ap) (&(ap))
      |                                  ~^~~~~~
      |                                   |
      |                                   __va_list_tag **
jsapi.c:267:39: note: in expansion of macro 'JS_ADDRESSOF_VA_LIST'
  267 |                                       JS_ADDRESSOF_VA_LIST(ap))) {
      |                                       ^~~~~~~~~~~~~~~~~~~~
jsapi.c:137:44: note: expected '__va_list_tag (*)[1]' but argument is of type '__va_list_tag **'
  137 |                      jsval **vpp, va_list *app)
      |                                   ~~~~~~~~~^~~
```

--
--- a/src/spidermonkey/js/src/jsapi.c
+++ b/src/spidermonkey/js/src/jsapi.c
@@ -93,7 +93,7 @@
 #ifdef HAVE_VA_LIST_AS_ARRAY
 #define JS_ADDRESSOF_VA_LIST(ap) ((va_list *)(ap))
 #else
-#define JS_ADDRESSOF_VA_LIST(ap) (&(ap))
+#define JS_ADDRESSOF_VA_LIST(ap) ((va_list *)(&(ap)))
 #endif

 #if defined(JS_PARANOID_REQUEST) && defined(JS_THREADSAFE)
